tools/ldb_cmd.cc

```diff
@@ -508,12 +524,52 @@ std::vector<std::string> LDBCommand::BuildCmdLineOptions(
                                   ARG_FIX_PREFIX_LEN,
                                   ARG_TRY_LOAD_OPTIONS,
                                   ARG_DISABLE_CONSISTENCY_CHECKS,
+                                  ARG_ENABLE_BLOB_FILES,
+                                  ARG_MIN_BLOB_SIZE,
+                                  ARG_BLOB_FILE_SIZE,
+                                  ARG_BLOB_COMPRESSION_TYPE,
+                                  ARG_ENABLE_BLOB_GARBAGE_COLLECTION,
+                                  ARG_BLOB_GARBAGE_COLLECTION_AGE_CUTOFF,
+                                  ARG_BLOB_GARBAGE_COLLECTION_FORCE_THRESHOLD,
+                                  ARG_BLOB_COMPACTION_READAHEAD_SIZE,
                                   ARG_IGNORE_UNKNOWN_OPTIONS,
                                   ARG_CF_NAME};
   ret.insert(ret.end(), options.begin(), options.end());
   return ret;
 }
 
+/**
+ * Parses the specific double option and fills in the value.
+ * Returns true if the option is found.
+ * Returns false if the option is not found or if there is an error parsing the
+ * value.  If there is an error, the specified exec_state is also
+ * updated.
+ */
+bool LDBCommand::ParseDoubleOption(
+    const std::map<std::string, std::string>& /*options*/,
+    const std::string& option, double& value,
+    LDBCommandExecuteResult& exec_state) {
+  std::map<std::string, std::string>::const_iterator itr =
+      option_map_.find(option);
+  if (itr != option_map_.end()) {
+    try {
+#if defined(CYGWIN)
+      value = std::strtod(itr->second.c_str(), 0);
+#else
+      value = std::stod(itr->second);
+#endif
```

Comment:
It is only `std::stod` that can throw here; `std::strtod` would communicate errors by setting `errno` to `ERANGE` or setting the optional `str_end` argument. So I think we would want to restructure this a bit like this:

```
#if defined(CYGWIN)
   char* str_end = nullptr;
   value = std::strtod(itr->second.c_str(), &str_end);
   if (str_end == itr->second.c_str()) {
      // handle invalid value
   } else if (errno == ERANGE) {
      // handle out of range
   } else {
      return true;
   }
#else
   try {
      value = std::stod(itr->second);
      return true;
   } catch (const std::invalid_argument&) {
      // handle invalid value
   } catch (const std::out_of_range&) {
      // handle out of range
   }
#endif
```

Reply:
This would actually also apply to `ParseIntOption`; it would be awesome if you could fix up that method as well.